src/Akka.Cluster.Hosting/AkkaClusterHostingExtensions.cs

```diff
@@ -13,38 +13,151 @@
 using Akka.Configuration;
 using Akka.Hosting;
 
+#nullable enable
 namespace Akka.Cluster.Hosting
 {
     /// <summary>
-    /// The set of options for enabling Akka.Cluster support.
+    ///     The set of options for enabling Akka.Cluster support.
     /// </summary>
     public sealed class ClusterOptions
     {
         /// <summary>
-        /// The akka.cluster.roles values.
+        ///     The akka.cluster.roles values.
         /// </summary>
-        public string[] Roles { get; set; }
+        public string[]? Roles { get; set; }
 
         /// <summary>
-        /// If populated, the akka.cluster.seed-nodes that will be used.
+        ///     Optional cluster role check to consider if a specific cluster role have enough
+        ///     members to be considered to be up. The default value is 1 node per role. 
         /// </summary>
-        public Address[] SeedNodes { get; set; }
+        public Dictionary<string, int>? MinimumNumberOfMembersPerRole { get; set; }
```

Comment:
This is the line I ment.
Would this be a better compromise to the Role class?
We're not forcing the user to provide any values and still keeps the class as simple as possible.